src/App.js

```diff
@@ -1,220 +1,173 @@
+          ...cell,
+          mode: mode.mode
+        };
       });
-      return newGrid;
-    }));
-
+    });
   }
 
   function handleMouseDown(cell) {
-    setMouseStatus({down:true});
-    setRange(prev=>({
-      start: {...cell},
-      end: null,
-    }));      
+    setRange({
+      start: { ...cell },
+      end: null
+    });
+    setMouseStatus({ down: true });
   }
 
   function handleMouseUp() {
-    
-    setGrid((prev=>{
-      const newGrid = prev.map(cell=>{
-        let newCell = {...cell};
+    setGrid((prev) => {
+      const newGrid = prev.map((cell) => {
+        let newCell = { ...cell };
         newCell.selected = false;
         return newCell;
       });
       return newGrid;
-    }));
-    setMouseStatus({down:false});
-    setRange({end:null, start:null});
-    
+    });
+    setMouseStatus({ down: false });
+    setRange({ end: null, start: null });
   }
   function handleMouseMove(cell) {
-   setRange(prev=>({
+    setRange((prev) => ({
       ...prev,
-      end:{...cell}
+      end: { ...cell }
     }));
   }
 
-  function handleChangeMode(e){
-    setMode({mode:e.target.value});
+  function handleModeChange(e) {
+    setMode({ mode: e.target.value });
   }
   function clearGrid() {
-    setGrid((prev=>{
-      const newGrid = prev.map(cell=>{
-        let newCell = {...cell};
+    setGrid((prev) => {
+      const newGrid = prev.map((cell) => {
+        let newCell = { ...cell };
         newCell.selected = false;
         newCell.mode = 'active';
         return newCell;
       });
       return newGrid;
-    }));
+    });
   }
-  
-
-  useEffect(()=>{
-    document.addEventListener("mouseup",handleMouseUp);
-    // document.addEventListener("click",handleMouseClick)
-    
-  },[])
-  useEffect(() => {
-    setGrid((prev=>{
-      const prevGrid = [...prev];
-      let newGrid = []
-      if (range.start && range.end) {
-      const start = range.start.id;
-      const end = range.end.id;
-        newGrid = prevGrid.map(cell=>{
-          const newCell = {...cell};
-          if(start <=end ) {
-            if(newCell.id >= start && newCell.id <= end){
-             newCell.selected = true;
-            }else {
-              newCell.selected = false;
-            }
-
-          } else if (end < start && mouseStatus.down) {
-            if(newCell.id >= end && newCell.id <= start){
-               newCell.selected = true;
-            }else{
-              newCell.selected = false;
-            }
-          }
-          
-          return newCell;
-        })
-        return newGrid;
-      }
-      return prevGrid;
-      
-    }));
-
+  function setRangeMinToMax(startID, endID) {
+    //function checks which one is bigger than the other, and returns => min, max
 
-    
-    
-    }, [range]);
+    if (startID > endID) {
+      return {
+        start: endID,
+        end: startID
+      };
+    }
+    return {
+      start: startID,
+      end: endID
+    };
+  }
+  function Row({ row }) {
```

Comment:
The reason for onClick not work is here:

In React, we cannot create components inside components. By doing that, React doesn't know this is a component and won't be able to keep its reference after state updates, which might cause bugs.

In this case the bug is that after onMouseDown is called and the state is updated, the onClick is gone because of the re-render.

The solution is to **always create components isolated**. Move Row to be outside `App` or execute this code directly in the render function below.